src/main/java/com/google/sps/utility/LocationsUtility.java

```diff
@@ -63,4 +64,39 @@ private static String getLocation(Pattern pattern, String prefix, String taskNot
     }
     return "";
   }
+
+  /**
+   * Generates combinations of the lists and appends them to the result parameter.
+   *
+   * @param lists A list of lists to generate combinations for.
+   * @param result A pointer to a list to contain the generated combinations.
+   * @param depth An int to represent the index of the list that is currently being traversed in the
+   *     recursive call.
+   * @param current A list containing the combination built so far at the recursive call.
+   */
+  private static void generateCombinationsHelper(
+      List<List<String>> lists, List<List<String>> result, int depth, List<String> current) {
```

Comment:
idea: `depth == current.size()` so you can drop that parameter if you want (but feel free to leave it if you think it makes things cleaner)

Reply:
I would actually prefer `depth = current.size()` and drop the parameter because then I don't have to worry about not checking the length on current in `current.remove(depth)`. Thanks!

Reply:
`RESOLVED`